public/app/features/plugins/admin/components/PluginList.tsx

```diff
@@ -23,9 +25,11 @@ export const PluginList = ({ plugins, displayMode }: Props) => {
 
   return (
     <div className={cx(styles.container, { [styles.list]: isList })} data-testid="plugin-list">
-      {plugins.map((plugin) => (
-        <PluginListItem key={plugin.id} plugin={plugin} pathName={pathName} displayMode={displayMode} />
-      ))}
+      {plugins
+        .filter((plugin) => contextSrv.hasAccessInMetadata(AccessControlAction.PluginsRead, plugin, true))
```

Comment:
Maybe I am just missing something, but do we need to have this filtering on the frontend side if the backend API is taking care of only returning plugins that the user can see?

Reply:
Currently we are not filtering the list returned by grafana-com in the backend. We are just adding the access metadata if I understand everything correct. But with that said it feels like we probably should move this filtering logic to the backend instead of having it here in the frontend.

Or would that introduce any edge cases that I'm not thinking about?